static/templyte/src/app/download-template/download-template.component.ts

```diff
@@ -68,6 +73,45 @@ export class DownloadTemplateComponent implements OnInit {
         this.authService.logout(null);
     }
 
+    storeRating(new_rating) {
+        const component = this;
+        const authorUID:string  =  component.user.uid;
+        component.templateRatingsInfoDatabaseRef.once('value').then(snapshot2 => {
+            const old_rating = snapshot2.val(); //value of previous rating
+            var varNumRatings = 0;
+            var varRatingSum = 0;
+            component.templateRatingsInfoDatabaseRef.set(
+                 new_rating
+            );
+            let hasRated:boolean = old_rating != null;
+            if (hasRated) {
+                component.templateDirectoryInfoDatabaseRef.child('/ratingSum').transaction(function(ratingSum){
+                    varRatingSum = ratingSum - old_rating + new_rating;
```

Comment:
Not much point of using a variable if you're going to do the calculation within the transaction anyway